packages/hooks/src/useTransferBetweenAccounts.ts

```diff
@@ -0,0 +1,143 @@
+import { useMemo } from 'react';
+import { useStore } from '@deriv/stores';
+import { useFetch } from '@deriv/api';
+import useActiveWallet from './useActiveWallet';
+import useActiveWalletCFDAccounts from './useActiveWalletCFDAccounts';
+import useCurrencyConfig from './useCurrencyConfig';
+import useWalletsList from './useWalletsList';
+
+const useTransferBetweenAccounts = () => {
+    const { ui } = useStore();
+    const { is_dark_mode_on } = ui;
+
+    const active_wallet = useActiveWallet();
+
+    const { data: wallets } = useWalletsList();
+
+    const { getConfig } = useCurrencyConfig();
+
+    const trading_apps_icon = is_dark_mode_on ? 'IcWalletOptionsDark' : 'IcWalletOptionsLight';
+
+    const {
+        data: { derivez_accounts, dxtrade_accounts, mt5_accounts },
+        is_success: is_cfd_accounts_loaded,
+    } = useActiveWalletCFDAccounts();
+
+    const { data, ...rest } = useFetch('transfer_between_accounts', {
+        payload: { accounts: 'all' },
+        options: { enabled: is_cfd_accounts_loaded },
+    });
+
+    const modified_transfer_accounts = useMemo(() => {
+        const all_linked_cfd_accounts = [...derivez_accounts, ...dxtrade_accounts, ...mt5_accounts];
+
+        const getAccountType = (is_demo?: number, currency?: string): 'fiat' | 'crypto' | 'demo' => {
+            if (is_demo) return 'demo';
+            return getConfig(currency || '')?.is_crypto ? 'crypto' : 'fiat';
+        };
+
+        const accounts = data?.accounts?.map(account => {
+            return {
+                ...account,
+                active_wallet_icon: active_wallet?.icon,
+                balance: parseFloat(
+                    Number(account.balance).toFixed(getConfig(account.currency || '')?.fractional_digits)
+                ),
+                display_currency_code: getConfig(account.currency || '')?.display_code,
+                is_demo: Boolean(account?.demo_account),
+                shortcode: active_wallet?.landing_company_name,
+                type: getAccountType(account.demo_account, account.currency),
+            };
+        });
+
+        return {
+            accounts:
```

Comment:
Since `account_category` based on RFC is either `wallet` or `trading`, I believe we can go with the same convention here as well.
```suggestion
            trading:
```